server/index.ts

```diff
@@ -1,40 +1,184 @@
 import ActorsRouter from './api/actors';
 import DirectorsRouter from './api/directors';
 import GenresRouter from './api/genres';
+import passport from 'passport';
+import passportLocal from 'passport-local';
+import cookieParser from 'cookie-parser';
+import session from 'express-session';
+import formData from 'express-form-data';
+const app = express();
+import type { Response, Request } from 'express';
 import TwitterRouter from './api/twitter';
 import UsersRouter from './api/users';
-import type { Request, Response } from 'express';
 
-const app = express();
+import { addUser, getUserById } from './database';
+import { profile } from 'console';
+import { any } from 'sequelize/dist/lib/operators';
+// const auth = require('./helpers/auth');
+// const authroutes = require('./api/authroutes');
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+
+
 const port = process.env.PORT || 3000;
 const dist = path.resolve(__dirname, '..', 'client/dist');
 app.use(cors({origin: true, credentials: true}));
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
 app.use(express.static(dist))
 app.use(express.json());
 
+
 app.use(cors({ credentials: true,
   'allowedHeaders': ['sessionId', 'Content-Type'],
   'exposedHeaders': ['sessionId'],
   'origin': '*',
   'methods': 'GET,HEAD,PUT,PATCH,POST,DELETE',
   'preflightContinue': false}));
 
-app.use(express.static(dist));
-app.use(bodyParser.json());
-app.use('/api/movies', MoviesRouter);
-app.use('/api/actors', ActorsRouter);
-app.use('/api/directors', DirectorsRouter);
-app.use('/api/genres', GenresRouter);
-app.use('/api/twitter', TwitterRouter);
-app.use('/api/users', UsersRouter);
+  app.use(express.static(dist));
+  app.use(bodyParser.json());
+  //Routers
+  app.use('/api/movies', MoviesRouter);
+  app.use('/api/actors', ActorsRouter);
+  app.use('/api/directors', DirectorsRouter);
+  app.use('/api/genres', GenresRouter);
+  app.use('/api/twitter', TwitterRouter);
+  app.use('/api/users', UsersRouter);
+
+
+  //Start of Passport
+  app.use(cookieParser());
+  app.use(formData.parse());
+
+  app.use(session({
+    secret: `${process.env.google_clientID}`,
+    saveUninitialized: false,
+    resave: true
+  }));
+
+
+  app.use(passport.initialize());
+  app.use(passport.session());
+
+// this is not getting used, but may be useful later when integrating the twitter strategy
+
+// app.use(auth.passport.initialize());
+// app.use(auth.passport.session());
+// app.use('/', authroutes);
+
+passport.use(new GoogleStrategy({
+  clientID: `${google_clientID}`,
+  clientSecret:`${google_clientSecret}`,
+  callbackURL: `${passportCallbackURL}`
+},
+  async function(request: any, accessToken: any, refreshToken: any, profile: any, done: any) {
+
+  //Find or Create a user
+  //returns [{}, boolean]
+  // [0] = user object
+  // [1] = true if created, false if found
+  const newUser = await User.findOrCreate({where: {email_Oauth: profile.id},
```

Comment:
this database function should be in `database/index.ts`

Reply:
Set database/index.ts addUser to use findOrCreate, functions as this did previously. Thanks for the tip!